kvbc/src/ReplicaImp.cpp

```diff
@@ -204,7 +204,7 @@ ReplicaImp::ReplicaImp(ICommunication *comm,
   auto dbSet = storageFactory->newDatabaseSet();
   m_bcDbAdapter = std::move(dbSet.dbAdapter);
   m_metadataDBClient = dbSet.metadataDBClient;
-
+  m_metadataDBClient->setAggregator(aggregator);
```

Comment:
I think we need to also set an aggregator to the `dataDBClient` from the `dbSet` as it is the one actually used for writing data for the blockchain itself.

Reply:
This is a shared_ptr. So it doesn't really matter.

Reply:
Yes, you are right. However, that assumes it points to the same object and it might not be the case for all implementations. Please correct me if I am missing something.

Reply:
But in the line before we set ` m_metadataDBClient = dbSet.metadataDBClient;`.
So it does point to the same object...
Am I missing somthing?

Reply:
I meant the other DB client from the set, i.e. `dbSet.dataDBClient` . They can be the same DB client (i.e. `dbSet.dataDBClient == dbSet.dbSet.metadataDBClient`) in case of a normal read-write replica or they can differ, i.e. for a RO replica that stores data on an object store (then the `dbSet.dataDBClient` will be an S3 client).

Reply:
done